### ISSUE TITLE
Use feature detection instead of platform detection if possible

### ISSUE THREAD

**Issue by Fjan** [nanoc/nanoc#1717]
The dart-sass filter is currently non-functional on some PCs due to a problem with the tty-platform gem that we use to detect whether we are on Windows. (https://github.com/nanoc/nanoc/issues/1703, https://github.com/piotrmurach/tty-platform/issues/1) 
While this will probably resolve itself in time, it's a bit brittle to rely on a platform string that might break as new versions of ruby are released. Instead I think it would be better to feature detect that we are on a Windows file system, like this:

```ruby
    def windows_fs?
      ::File::ALT_SEPARATOR == "\\"
    end
```

At the moment the dart-sass filter seems to be the only place in nanoc we use this windows detection, I think? For detecting other windows related properties we may need to add additional helper functions, but in general I think this would be preferable to depending on a platform to be reliably detected and to infer functionality from that.

**Comment by denisdefreyne** [comment]
There is a possible fix in #1718. Can you check whether this works for you?

**Comment by Fjan** [comment]
Thanks, looks good to me. I don't have access to a PC with git on it though, so can only verify it after release I'm afraid.
